mautrix_telegram/web/provisioning/__init__.py

```diff
@@ -528,6 +531,36 @@ async def retry_takeout(self, request: web.Request) -> web.Response:
         user.takeout_retry_immediate.set()
         return web.json_response({}, status=200)
 
+    async def login_qr(self, request: web.Request) -> web.Response:
+        _, user, err = await self.get_user_request_info(request, websocket=True)
+        if err is not None:
+            return err
+
+        await user.ensure_started(even_if_no_session=True)
+        qr_login = QRLogin(user.client, ignored_ids=[])
+
+        ws = web.WebSocketResponse(protocols=["net.maunium.telegram.login"])
+        await ws.prepare(request)
+
+        user_info = None
+        try:
+            await qr_login.recreate()
+            await ws.send_json({"code": qr_login.url, "timeout": 30})
```

Comment:
Is the timeout always 30? Might be better to get the real timeout instead of hardcoding it (something like `(qr_login.expires - datetime.datetime.now(tz=datetime.timezone.utc)).total_seconds()`)